File: eclipse-plugins/edu.wpi.first.wpilib.plugins.java/resources/templates/examples/GearsBot/src/org/usfirst/frc/team190/robot/commands/OpenClaw.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package $package.commands;

import edu.wpi.first.wpilibj.command.Command;
import $package.Robot;

/**
 * Opens the claw for one second. Real robots should use sensors, stalling
 * motors is BAD!
 */
public class OpenClaw extends Command {
    
    public OpenClaw() {
        requires(Robot.claw);
        setTimeout(1);
    }

    // Called just before this Command runs the first time
    protected void initialize() {
        Robot.claw.open();
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return isTimedOut();
    }

    // Called once after isFinished returns true
    protected void end() {
        Robot.claw.stop();
    }

    // Called when another command which requires one or more of the same
    // subsystems is scheduled to run
    protected void interrupted() {
        end();
    }
}

File: eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/GearsBot/src/Commands/OpenClaw.cpp
#include "OpenClaw.h"
#include "Robot.h"

OpenClaw::OpenClaw() : Command("OpenClaw") {
	Requires(Robot::claw.get());
	SetTimeout(1);
}

// Called just before this Command runs the first time
void OpenClaw::Initialize() {
	Robot::claw->Open();
}

// Called repeatedly when this Command is scheduled to run
void OpenClaw::Execute() {}

// Make this return true when this Command no longer needs to run execute()
bool OpenClaw::IsFinished() {
	return IsTimedOut();
}

// Called once after isFinished returns true
void OpenClaw::End() {
	Robot::claw->Stop();
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void OpenClaw::Interrupted() {
	End();
}
